binary_search_tree/binary_search_tree.py

```diff
@@ -11,22 +12,22 @@ def __init__(self, value):
     Once a leaf node is found,
     the new node is added as a child of the leaf node.
     """
+    # O(log n)
     def insert(self, value):
-        
         # Defines wich side of the tree to add the value as the string version of the attribute
-        attr = "right" if value >= self.value else "left"
+        attr = "right" if value >= self.value else "left"  # O(1)
         # Gets the child binary search tree desired (either larger or smaller side)
-        child_bst = getattr(self, attr)
+        child_bst = getattr(self, attr)  # O(1) probably
         # If the child is empty
         if child_bst is None:
             # create a new bst
-            new_bst = BinarySearchTree(value)
+            new_bst = BinarySearchTree(value)  # O(1)
             # set it to the new bst
-            setattr(self, attr, new_bst)
+            setattr(self, attr, new_bst)  # O(1) probably
         # If it's not
         else:
             # Insert it into child tree (recursively)
-            child_bst.insert(value)
+            child_bst.insert(value)  # Recursing through half so O(log n)
```

Comment:
And you don't have to specify right or left by calling it on the child you just referenced.